### ISSUE TITLE
Inline code output is not interpreted as Markdown anymore

### ISSUE THREAD

**Issue by salim-b** [quarto-dev/quarto-cli#10197]
### Bug description

[Inline code](https://quarto.org/docs/computations/inline-code.html) output from the knitr engine (I didn't test other engines) is not interpreted as Markdown anymore by Quarto 1.5.52. This is a regression from Quarto 1.4.x.

### Steps to reproduce

1. Copy the following content into a new file `test.qmd`

    ````rmd
    ---
    title: "test"
    engine: knitr
    ---
    
    ## Quarto
    
    A [hyperlink](https://quarto.org).
    
    The same `{r} '[hyperlink](https://quarto.org)'` from inline code is not rendered as Markdown.
    
    Wrapping the `{r} knitr::asis_output('[hyperlink](https://quarto.org)')` in `knitr::asis_output()` doesn't help.
    ````

1. Render the document to HTML:

   ```sh
   quarto render test.qmd
   ```

### Expected behavior

Inline R code output should be interpreted as Markdown.

### Actual behavior

The inline R code output (`[hyperlink](https://quarto.org)`) is not interpreted as Markdown but printed as-is:

![Screenshot 2024-07-02 at 17-20-44 test](https://github.com/quarto-dev/quarto-cli/assets/20040931/ee958bf8-d0ed-47fc-8d4e-d98759ac06b8)

### Your environment

OS: Ubuntu 22.04

### Quarto check output

```
Quarto 1.5.52
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.52
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /home/salim/.local/bin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /usr/lib/R
      LibPaths:
        - /home/salim/.software/managed/r
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.47
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```

**Comment by cscheid** [comment]
> This is a regression from Quarto 1.4.x.

Sorry, but that doesn't seem right. This is the same behavior as in 1.4.557:

<img width="1237" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/ad346634-8b26-4bc8-a1cc-2008247874da">

**Comment by cscheid** [comment]
In addition, if you wrap the string in `I()`, [as we explain in the documentation](https://quarto.org/docs/computations/inline-code.html#markdown-output), you get the correct output:

````
---
title: "test"
engine: knitr
---

## Quarto

A [hyperlink](https://quarto.org).

The same `{r} I('[hyperlink](https://quarto.org)')` from inline code is not rendered as Markdown.

Wrapping the `{r} knitr::asis_output('[hyperlink](https://quarto.org)')` in `knitr::asis_output()` doesn't help.
````

<img width="928" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/dd539d05-7516-4cfc-be7f-716a4b9fad7a">

**Comment by salim-b** [comment]
> Sorry, but that doesn't seem right. This is the same behavior as in 1.4.557:

Ah, sorry, then this regression was introduced earlier. At least with Quarto 1.3.450, this used to work.

Is there a recommended way produce Markdown output from inline code in current stable Quarto?

**Comment by cscheid** [comment]
> Ah, sorry, then this regression was introduced earlier. At least with Quarto 1.3.450, this used to work.

You wrote:

> This is a regression from Quarto 1.4.x.

@salim-b please, I have to ask you to make an effort to be accurate when you're writing in your issues, it saves us a lot of time in wild-goose chases.

**Comment by salim-b** [comment]
> please, I have to ask you to make an effort to be accurate when you're writing in your issues, it saves us a lot of time in wild-goose chases.

Sure, I usually try my best to be accurate. I observed the above issue while upgrading a project from Quarto 1.3.450 to the latest version (where I intermediately updated to 1.4.x first, which already brought huge changes and needed a lot of adaptions from user side), so I confused the Quarto versions (I mistakenly thought I didn't see the issue with 1.4.x). Sorry again for this mistake.

Only from your answer I learned about the difference between `` `r something` `` and `` `{r} something` `` regarding Markdown interpretation, which clears things up. Thanks again!
